### PR TITLE
Doc enhancements

### PR DESCRIPTION
**Pull request by rosemarymarano**
Working with @tbarnes-us to update and improve WKO docs. First pass at Home page, add contributing info to the Developer Guide, and added FAQ from the wiki to the doc FAQ (Answers to commonly asked newcomer questions).

### CODE REVIEW

File: documentation/staging/content/_index.md
Comment (rjeberhard) [inline documentation/staging/content/_index.md:3]: We should make the same set of changes to the top-level README.md, which is the content displayed when you visit the GitHub project page.

File: documentation/staging/content/faq/newbie.md
Comment (rjeberhard) [inline documentation/staging/content/faq/newbie.md:58]: We've also certified Calico.

File: documentation/staging/content/faq/newbie.md
Comment (rjeberhard) [inline documentation/staging/content/faq/newbie.md:44]: In other places, we've moved away from highlighting Docker to mentioning "container images" or running in "containers". Many if not most will understand that Docker support means that we also support other container runtimes, such as cri-o, but we make this clearer by no longer preferencing Docker.

File: documentation/staging/content/_index.md
Comment (rosemarymarano) [reply documentation/staging/content/_index.md:3]: Will do.

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:58]: I'll add it.

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:44]: I'll change it.

File: documentation/staging/content/faq/newbie.md
Comment (tbarnes-us) [inline documentation/staging/content/faq/newbie.md:63]: To me, the blog reference looks a bit too low level for this answer.  How about something like "The operator will automatically create Kubernetes services for SSL ports for you, just like it does for non-SSL ports. You will need to supply your domain's SSL certs and SSL configuration (just like on-premise); this is typically accomplished by including the certs in a Kubernetes config map, mounting the config map to your pods [link to FAQ entry that describes how to mount a configMap to a domain], and modifying your SSL configuration to reference the mount location."

@rjeberhard

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [inline documentation/staging/content/faq/newbie.md:64]: @rjeberhard This question seems to be Docker specific. I'm inclined to leave it as such unless you tell me otherwise.

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [inline documentation/staging/content/faq/newbie.md:122]: @rjeberhard This answer centers on Docker. I'm not sure how to generalize it correctly. Your input would help.

File: documentation/staging/content/faq/newbie.md
Comment (tbarnes-us) [inline documentation/staging/content/faq/newbie.md:93]: Is this blog up to date? It's dated 2017.

File: documentation/staging/content/faq/newbie.md
Comment (tbarnes-us) [inline documentation/staging/content/faq/newbie.md:96]: Is this blog up-to-date? It's dated 2017.

File: documentation/staging/content/faq/newbie.md
Comment (tbarnes-us) [inline documentation/staging/content/faq/newbie.md:107]: Add a reference to the life-cycle doc.

File: documentation/staging/content/faq/newbie.md
Comment (tbarnes-us) [inline documentation/staging/content/faq/newbie.md:121]: This "See" reference is out of date. Instead, reference the new patching chapter, the life-cycle doc, and the CI/CD doc...  I assume @mriccell or @rjeberhard may also want to word-smith the contents of the answer.

File: documentation/staging/content/faq/newbie.md
Comment (tbarnes-us) [inline documentation/staging/content/faq/newbie.md:129]: @rjeberhard I'm not familiar with security, but this answer looks confusing to me. It seems to be discussing the operator's certs and not WL SSL certs in general?

File: documentation/staging/content/faq/newbie.md
Comment (tbarnes-us) [inline documentation/staging/content/faq/newbie.md:137]: are persisted --> are optionally persisted

Comment (tbarnes-us) [reply documentation/staging/content/faq/newbie.md:137]: Or in more detail:  WebLogic Server stdout logging is echoed to their pod logs by default, and WebLogic Server file logs are optionally persisted to an external volume.

Comment (tbarnes-us) [reply documentation/staging/content/faq/newbie.md:137]: Also, I think there's an Elastic stack sample now that can be linked here?

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:63]: @tbarnes-us I'll change the answer as you prescribe (but, certs=certificates?)

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:96]: Doubt it. These FAQ are old as are their references. If the blog is no longer applicable, we should the delete the reference to it.

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:107]: Will do.

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:44]: Done. You'll see it in the next commit.

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:93]: Not sure. We can ask @lennyphan

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:137]: Got it.

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:121]: @tbarnes-us I can't x-ref the new patching section until it's in the docs, but I added the other two. If @mriccell or @rjeberhard approves the patching doc, I'll add it.

File: documentation/staging/content/faq/newbie.md
Comment (rjeberhard) [reply documentation/staging/content/faq/newbie.md:129]: This answer is specifically about the operator; however, I agree that it's confusing because there is no context. I'd be happy to just remove it.

File: documentation/staging/content/faq/newbie.md
Comment (rjeberhard) [reply documentation/staging/content/faq/newbie.md:63]: Tom's answer is really good, but yes, I'd use the complete word for "certificates."

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:129]: Works for me. Done in the next push.

File: documentation/staging/content/faq/newbie.md
Comment (rjeberhard) [reply documentation/staging/content/faq/newbie.md:64]: Let's remove this question and answer too.... By the way, where did these come from? Many are really useful, but then others sound "ancient."

File: documentation/staging/content/faq/newbie.md
Comment (rjeberhard) [reply documentation/staging/content/faq/newbie.md:121]: As above, I'm tempted to remove the question or at least the content here and just point to the new patching documentation. Which one do I need to approve (overwhelmed today)?

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:64]: Some are. They came from the wiki FAQ, https://github.com/oracle/weblogic-kubernetes-operator/wiki/FAQ

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:121]: @rjeberhard https://github.com/oracle/weblogic-kubernetes-operator/pull/2328
(Sorry for the overload.)

File: documentation/staging/content/faq/newbie.md
Comment (tbarnes-us) [inline documentation/staging/content/faq/newbie.md:54]: The above is out of date.  How about:  

**Q:** How is location transparency achieved and the communication between WLS instances handled?

Inside the Kubernetes cluster, the operator generates a Kubernetes ClusterIP service for each WebLogic server pod called `DOMAINUID-WEBLOGICSERVERNAME` and for each WebLogic cluster called `DOMAINUID-cluster-CLUSTERNAME`. The operator also overrides your WebLogic network listen address configuration to reflect the service names so that you don't need to do this yourself. The services act as DNS names, and allow the pods with WebLogic Servers to move to different nodes in the Kubernetes cluster and continue communicating with other servers.  

**Q:** How is communication from outside the Kubernetes cluster handled?

HTTP communication to your applications from locations outside the cluster is typically accomplished by deploying a load balancer that redirects traffic to your domain's Kubernetes services (the operator automatically deploys these services for you). For example, [link to sample that sets up Traefik+ingress to a cluster - such as quick  start.]

RMI communication with locations outside of the Kubernetes cluster is typically accomplished by deploying a load balancer or using Kubernetes Node Ports, see [external client FAQ].

WebLogic console communication is accomplished using either a Kubernetes NodePort service for the WebLogic admin server that the operator can automatically deploy for you [link to doc or sample for doing this], or by configuring a load balancer to redirect console traffic [link to doc or sample for doing this] to your WebLogic admin server.

WebLogic remote console communication is accomplished by using a load balancer, see [link to doc or sample for doing this].

File: documentation/staging/content/faq/newbie.md
Comment (tbarnes-us) [inline documentation/staging/content/faq/newbie.md:10]: This (and any similar URLs) can and should be converted to relrefs.

Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:10]: Done. All the others are done also.

File: documentation/staging/content/_index.md
Comment (tbarnes-us) [inline documentation/staging/content/_index.md:10]: Add first bullet:
* Deploy an operator that manages all WebLogic domains in all namespaces in a Kubernetes cluster, or that only manages domains in a specific subset of the namespaces, or that only manages domains that are located in the same namespace as the operator. A namespace can be managed by at most one operator.

File: documentation/staging/content/_index.md
Comment (tbarnes-us) [inline documentation/staging/content/_index.md:7]: The operator uses a common set --> The operator leverages the [Kubernetes operator pattern]( https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) which means that it uses

File: documentation/staging/content/faq/newbie.md
Comment (rjeberhard) [reply documentation/staging/content/faq/newbie.md:114]: This looks good to me now.

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:54]: @tbarnes-us Not 100% sure about the links you want here:
1. Using either a Kubernetes NodePort service for the WebLogic admin server 
2. Configuring a load balancer to redirect console traffic
For 1, the QS guide, step 7, https://oracle.github.io/weblogic-kubernetes-operator/quickstart/create-domain/
For 2. ???, https://oracle.github.io/weblogic-kubernetes-operator/samples/simple/azure-kubernetes-service/domain-on-pv/#deploy-sample-application

File: documentation/staging/content/faq/newbie.md
Comment (tbarnes-us) [reply documentation/staging/content/faq/newbie.md:10]: I don't see the the change. Did you push it?

File: documentation/staging/content/faq/newbie.md
Comment (lennyphan) [reply documentation/staging/content/faq/newbie.md:88]: This blog is out of date.  The WLDF description is mostly valid for the rules but the use of webhooks pre-dates the operator's REST API.

File: documentation/staging/content/_index.md
Comment (rosemarymarano) [reply documentation/staging/content/_index.md:7]: got it.

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:10]: Not yet, a few more to go (waiting for the edit traffic to slow down; I think it's coming).

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:88]: Thanks for the review. I'll remove the reference and, perhaps, point them to the Scaling section in the docs, https://oracle.github.io/weblogic-kubernetes-operator/userguide/managing-domains/domain-lifecycle/scaling/ Does that work for you?

Comment (lennyphan) [reply documentation/staging/content/faq/newbie.md:88]: Yes, this is much better link.  Thank you!

Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:88]: Done.

File: documentation/staging/content/faq/newbie.md
Comment (tbarnes-us) [reply documentation/staging/content/faq/newbie.md:54]: Geez - it looks like we have outages here. There's a new chapter for 'remote console' that's somewhat helpful, but nothing for 'WebLogic console' that I can see except buried too deep in the samples.  

For now:
- Access to arbitrary HTTP apps through a LB: point to the 'ingress' chapter.
- Access to the WL console through a LB: maybe point to the MII sample? 
- WebLogic console through a NP: mention 'domain.spec.adminServer.adminService.channels' and point to the domain resource doc 

For the latter two we should have a dedicated chapter just like we now have for the remote console.

Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:54]: We'll get those docs in the next round.

File: documentation/staging/content/faq/newbie.md
Comment (tbarnes-us) [inline documentation/staging/content/faq/newbie.md:63]: Can we organize the answer a little differently? For example:

```
**Q:** How is communication from outside the Kubernetes cluster handled?

**A:** 

* _HTTP communication to your applications from locations outside the cluster:_  This is typically accomplished by deploying a load balancer that redirects traffic to your domain's Kubernetes services (the operator automatically deploys these services for you).  See [Ingress]({{< relref "/userguide/managing-domains/ingress/_index.md" >}}).
For example, see the Quick Start, [Install the operator and ingress controller]({{< relref "/quickstart/install.md" >}}).
* _JMS, EJB, and other types of RMI communication with locations outside of the Kubernetes cluster:_  This is typically accomplished by tunneling the RMI traffic over HTTP through a load balancer or, less commonly accomplished by using T3 or T3S directly with Kubernetes Node Ports. See [External WebLogic clients]({{< relref "/faq/external-clients.md" >}}).
* _Access the WebLogic Server Administration Console_:  This can be done through a load balancer, see the [Model in Image]({{< relref "/samples/simple/domains/model-in-image/_index.md" >}}) sample.  Or this can be done through a Kubernetes NodePort service, run `$ kubectl explain domain.spec.adminServer.adminService.channels`.
* _Access the WebLogic Server Remote Console_: This can be done using a load balancer or Kubernetes NodePort service. See [Use the Remote Console]({{< relref "/userguide/managing-domains/accessing-the-domain/admin-console.md" >}}).
```

**Review by tbarnes-us — APPROVED** [review]
LGTM - I'm approving, and I've added one last FAQ comment in case you want to act on it.

File: documentation/staging/content/faq/newbie.md
Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:63]: Yes, 100% better. Done.

**Comment by rosemarymarano** [comment]
Absolutely incorporated your comment/improvement. I think everything looks good, but we cannot merge this PR until after https://github.com/oracle/weblogic-kubernetes-operator/pull/2328 due to a link in here to the patching doc there. @rjeberhard

File: documentation/staging/content/faq/newbie.md
Comment (tbarnes-us) [inline documentation/staging/content/faq/newbie.md:121]: Placeholder: Missing relref for `[Apply patched images to a running domain] ` (which I assume is intended to reference the new patching document that coming in pull 2328).

Comment (rosemarymarano) [reply documentation/staging/content/faq/newbie.md:121]: Correct.

**Review by rjeberhard — APPROVED** [review]

**Pull request merged**
